packet/flatcar-linux/kubernetes/dns.tf

```diff
@@ -0,0 +1,34 @@
+module "dns-entries" {
```

Comment:
We could instantiate both `route53` and `manual` and have a knob for which DNS provider should be used. Something like
```
entries = var.dns_provider == "route53" ? local.dns_entries : 0
```

This way user does not have to modify the module, which I believe should  be discouraged.

The downside is that at some point we will have more modules, which will be empty, but this is kind of Terraform limitation... But I think then we can take this outside of kubernetes module and let user define them and drive them using `-target`.

Reply:
Then route53 would require to get some dummy credentials?

Reply:
Oh, it seems `aws` provider does not lazily evaluate the credentials :(

```
provider "aws" {
  version = "2.31.0"

  region = "us-east-1"
}

resource "aws_route53_record" "dns-records" {
  count = 0

  type = "A"
  zone_id = "nonexistent"
  name = ""
}
```

Following code requires AWS credentials :man_facepalming:

Reply:
Yes, unfortunately it's a problem I haven't been able to find a feasible solution. There are two Terraform limitations that make this impossible to implement in a clean way now:
1. `module` does not support the count parameter: https://github.com/hashicorp/terraform/issues/953.
2. It's not possible to use a variable in the source of a module https://github.com/hashicorp/terraform/issues/1439.

Reply:
```
variable "use_aws" {
	default = 0
}

provider "aws" {
  version = "2.31.0"

  region = var.use_aws == 0 ? "nonexisting" : "us-east-1"
  skip_region_validation = var.use_aws == 0 ? true : false
  skip_credentials_validation = var.use_aws == 0 ? true : false
  skip_requesting_account_id = var.use_aws == 0 ? true : false
  skip_metadata_api_check = var.use_aws == 0 ? true : false
  skip_get_ec2_platforms = var.use_aws == 0 ? true : false
  access_key = var.use_aws == 0 ? "nonexistent" : null
  secret_key = var.use_aws == 0 ? "nonexistent" : null
}

resource "aws_route53_record" "dns-records" {
  count = var.use_aws

  type = "A"
  zone_id = "nonexistent"
  name = ""
}
``` Seems to work :see_no_evil:

Reply:
Note, to test that, one can use `docker run -it --entrypoint /bin/sh --rm hashicorp/terraform`.